cmd/dp-recipe-api/main.go

```diff
@@ -64,4 +67,36 @@ func main() {
 		api.CreateAndInitialiseRecipeAPI(ctx, *cfg, store.DataStore{Backend: nil})
 	}
 
+	apiErrors := make(chan error, 1)
+
+	// Gracefully shutdown the application closing any open resources.
+	gracefulShutdown := func() {
+		log.Event(ctx, (fmt.Sprintf("shutdown with timeout: %s", cfg.GracefulShutdownTimeout)), log.INFO)
+		ctx, cancel := context.WithTimeout(context.Background(), cfg.GracefulShutdownTimeout)
```

Comment:
The usage for the shutdown context timeout is not quite right. I would expect the context to be declared outside of the go routine, and then a wait on the done channel `<-ctx.Done()` 

See here for an example: https://github.com/ONSdigital/dp-dataset-api/blob/develop/main.go#L167

Reply:
Shutdown Context Timeout successfully Done in PR https://github.com/ONSdigital/dp-recipe-api/pull/127